llvm/include/llvm/CodeGen/GCMetadata.h

```diff
@@ -151,15 +152,47 @@ class GCFunctionInfo {
   size_t live_size(const iterator &p) const { return roots_size(); }
 };
 
-struct GCStrategyMap {
-  StringMap<std::unique_ptr<GCStrategy>> StrategyMap;
+class GCStrategyMap {
+  using MapT =
+      MapVector<std::string, std::unique_ptr<GCStrategy>, StringMap<unsigned>>;
```

Comment:
Where do these names come from? Can this use StringRef? Cursory search seems to suggest these are already live in the LLVMContext